src/Events/Controllers/AppEventsController.cs

```diff
@@ -217,6 +215,29 @@ public async Task<ActionResult<string>> Post([FromBody] CloudEventRequestModel c
             }
         }
 
+        private static void ValidateQueryParams(string after, DateTime? from, DateTime? to, int size)
+        {
+            if (string.IsNullOrEmpty(after) && from == null)
+            {
+                throw new ArgumentException("From or after must be defined.");
```

Comment:
Could we handle this with a return value instead? null if nothing is wrong, a string with an error message if validation fails.

Another, bigger, alternative would be to use ModelState. Collect the parameters in a model and have validation of the model being performed by the model binder. Then just check ModelState.IsValid.

Reply:
Tested ModelState, but not happy with the return object. Kept the code in a seperate branch so it can be revisited at a later point. Introducing a touple with validationstatus + error message for now.